packages/proxy/lib/http/util/prerequests.ts

```diff
@@ -157,8 +160,23 @@ export class PreRequests {
   }
 
   addPending (browserPreRequest: BrowserPreRequest) {
-    metrics.browserPreRequestsReceived++
     const key = `${browserPreRequest.method}-${tryDecodeURI(browserPreRequest.url)}`
+
+    // The initial request that loads the service worker does not always get sent to CDP. Thus, we need to explicitly ignore it. We determine
+    // it's the service worker request via the `service-worker` header
+    if (browserPreRequest.headers?.['Service-Worker'] === 'script') {
+      debugVerbose('Ignoring service worker script since we are not guaranteed to receive it', key)
+
+      return
+    }
+
+    if (this.serviceWorkerManager.processBrowserPreRequest(browserPreRequest)) {
+      debugVerbose('Not correlating request since it is fully controlled by service worker and the correlation will happen within the service worker', key)
```

Comment:
```suggestion
      debugVerbose('Not correlating request since it is fully controlled by the service worker and the correlation will happen within the service worker', key)
```

Reply:
Updated: https://github.com/cypress-io/cypress/pull/28517/commits/956dd0aa92c8e7cdb0ee1284be92bdeeb52426a3